parser.cc

```diff
@@ -487,6 +487,11 @@ class IteratorAdaptor : public std::iterator<std::bidirectional_iterator_tag, Ou
 			--s;
 			return *this;
 		}
+		inline IteratorAdaptor<Src, Out,In> &operator++(int)
```

Comment:
This should add the parameter to s, rather than just incrementing it.  It is possible to explicitly call operator++ with the parameter (though a horrible and evil thing to do).